1658144661198-WorkspaceInstanceUsageAddExtraFields.ts

```diff
@@ -0,0 +1,26 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+import {columnExists} from "@gitpod/gitpod-db/lib/typeorm/migration/helper/helper";
+
+export class WorkspaceInstanceUsageAddExtraFields1658144661198 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+
+        const columns = ['userId', 'workspaceId', 'projectId', 'workspaceType', 'workspaceClass'];
+
+        const statements = columns
+            .filter(async col => {
+                const exists = await columnExists(queryRunner, "d_b_workspace_instance_usage", col)
+                return !exists;
+            })
+            .map(col => {
+                return `ADD COLUMN ${col} varchar(255) NOT NULL DEFAULT ''`;
+            });
+
+        if (statements.length > 0) {
+            await queryRunner.query(`ALTER TABLE \`d_b_workspace_instance_usage\` ${statements.join(", ")};`);
+        }
```

Comment:
Note: I think statements here may _not_ end with `;` and will fail if they do for reasons unknown to mankind. Please double check what other migrations do. :+1:

Reply:
https://www.sqltutorial.org/sql-add-column/ says it should always end with a `;`. However `1649107789640-PrebuildStatusVersionColumn.ts` doesn't have a semicolon. I'll drop mine.

Reply:
Jep, it's a typeorm thing :grimacing: